src/nat64/tayga_helper.cpp

```diff
@@ -0,0 +1,128 @@
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if OTBR_ENABLE_NAT64
+#include <fstream>
+
+#include <openthread/border_router.h>
+
+#include "common/code_utils.hpp"
+#include "common/types.hpp"
+#include "nat64/tayga_helper.hpp"
+#include "utils/system_utils.hpp"
+
+namespace otbr {
+namespace Tayga {
+namespace {
+
+static const char kTaygaConf[]       = "/etc/tayga.conf";
+static const char kTaygaConfTmp[]    = "/etc/tayga_tmp.conf";
+static const char kTaygaRestartCmd[] = OTBR_TAYGA_RESTART_CMD;
+static const char kPrefixLineStart[] = "prefix ";
+
+bool UpdatePrefix(const std::string &prefix)
+{
+    bool          is_copied  = false;
+    bool          is_updated = false;
+    std::ifstream file(kTaygaConf);
+    std::ofstream tmp_file(kTaygaConfTmp);
+    std::string   line;
+
+    VerifyOrExit(file && tmp_file);
+
+    while (getline(file, line))
+    {
+        if (line.find(kPrefixLineStart, 0) == 0)
+        {
+            line       = kPrefixLineStart + prefix;
+            is_updated = true;
+        }
+        tmp_file << line << std::endl;
+        VerifyOrExit(tmp_file.good(), is_updated = false);
+    }
+
+    VerifyOrExit(file.eof(), is_updated = false);
+
+    file.close();
+    tmp_file.close();
+    is_copied = true;
+
+    if (is_updated)
+    {
+        VerifyOrExit(remove(kTaygaConf) == 0 && rename(kTaygaConfTmp, kTaygaConf) == 0, is_updated = false);
```

Comment:
```suggestion
        VerifyOrExit(rename(kTaygaConfTmp, kTaygaConf) == 0, is_updated = false);
```
`remove` not necessary?

Reply:
`rename` might fail for exiting files in some system. So call `remove` to make it safer.

Reply:
> `rename` might fail for exiting files in some system. So call `remove` to make it safer.

According to [manpages](https://linux.die.net/man/3/rename), the new file is always removed if exists. Do we have any clue why it may fail for existing files in some system?

Reply:
Based on the description of [rename()](https://www.cplusplus.com/reference/cstdio/rename/), `If newname names an existing file, the function may either fail or override the existing file, depending on the specific system and library implementation`. So I think first removing the existing file is safer?

Reply:
Thanks for the reference. LGTM.